server/controllers/controller_helpers/metrics.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'lodash';
+
+import responsesService from '../../db/services/responses';
+import campaignsService from '../../db/services/campaigns';
+
+function splitResponses(responseString) {
+  return responseString.split(',');
+}
+
+export function mapCallIds(callObjArray) {
+  return callObjArray.map(call => call.attributes.id);
+}
+
+export function checkCampaignNotDraft(campaignId, res) {
+  return campaignsService.getCampaignById({ id: campaignId })
+    .then((campaign) => {
+      if (campaign) {
+        const { status } = campaign.attributes;
+        if (status === 'draft') {
+          throw new Error('Campaign metrics not available for campaigns still in draft.');
+        }
+        return true;
+      }
+      return res.status(500).json({ message: 'problem with db query for campaign by id' });
+    })
+    .catch((err) => {
+      const errorMessage = `could not find campaigin by id in campaign status check (metrics): ${err}`;
+      return res.status(404).json({ message: errorMessage });
+    });
+}
+
+function createStatusMap(metricsObj) {
+  const statusOptions = ['AVAILABLE', 'ASSIGNED', 'IN_PROGRESS', 'HUNG_UP', 'ATTEMPTED'];
+  const statusMetrics = metricsObj.status_distribution;
+  statusOptions.forEach((statusOption) => {
+    statusMetrics[statusOption] = 0;
+  });
+  return metricsObj;
+}
+
+function createOutcomeMap(metricsObj) {
+  const outcomeOptions = ['PENDING', 'ANSWERED', 'BAD_NUMBER', 'DO_NOT_CALL', 'LEFT_MSG', 'NO_ANSWER', 'INCOMPLETE'];
+  const outcomeMetrics = metricsObj.outcome_distribution;
+  outcomeOptions.forEach((outcomeOption) => {
+    outcomeMetrics[outcomeOption] = 0;
+  });
+}
+
+function createResponsesMap(responseArray, questionId, metricsObj) {
+  const responseStats = metricsObj.response_stats;
+  responseStats[questionId] = {};
+  const responsesForQuestion = responseStats[questionId];
+  responseArray.forEach((responseOption) => {
+    if (responseOption !== '') {
+      responsesForQuestion[responseOption] = 0;
+    } else {
+      delete responseStats[questionId];
+    }
+  });
+  return metricsObj;
+}
+
+export function countStatusAndOutcomeFrequency(callsArray, metricsObj) {
+  createStatusMap(metricsObj);
+  createOutcomeMap(metricsObj);
+  callsArray.forEach((call) => {
+    const { status, outcome } = call.attributes;
+    const statusStats = metricsObj.status_distribution;
+    const outcomeStats = metricsObj.outcome_distribution;
+    if (Object.prototype.hasOwnProperty.call(statusStats, status)) {
```

Comment:
Not 100% sure, but just want to make sure this is the best way to do this check and that there isn't a less verbose way of accomplishing this? Applies to multiple lines in this file.

Reply:
there prob is, i think in the interest of time I'm going to say I can attempt to get back to this later, or it will be covered in the controller clean up we're doing. Does that sound fair? Just bc this is blocking Duy